pytket/qir/conversion/api.py

```diff
@@ -77,9 +80,26 @@ def pytket_to_qir(
     populated_module = qir_generator.circuit_to_module(
         qir_generator.circuit, qir_generator.module, True
     )
-    if returntype == ReturnTypeQIR.BINARY:
-        return populated_module.module.bitcode()
-    elif returntype == ReturnTypeQIR.STRING:
-        return populated_module.module.ir()
+
+    if pyqir_0_7_compatibility:
+
+        initial_result = str(populated_module.module.ir())  # type: ignore
+
+        result = initial_result.replace("entry_point", "EntryPoint").replace("num_required_qubits", "requiredQubits").replace("num_required_results", "requiredResults")  # type: ignore
+
+        bitcode = pyqir.Module.from_ir(pyqir.Context(), result).bitcode  # type: ignore
+
+        if returntype == ReturnTypeQIR.BINARY:
+            return bitcode  # type: ignore
+        elif returntype == ReturnTypeQIR.STRING:
+            return result  # type: ignore
+        else:
+            raise ValueError("unsupported return type")
```

Comment:
I think we can actually have an `assert` here, since we know these are the only values of the enum.

Reply:
done @cqc-melf
[replace valuerror for qir_format qith assertion](https://github.com/CQCL/pytket-qir/pull/35/commits/bbed6f40ababe52a4e1e8ef1176f1abe4dceeb71)
[bbed6f4](https://github.com/CQCL/pytket-qir/pull/35/commits/bbed6f40ababe52a4e1e8ef1176f1abe4dceeb71)